Andy Scott-Lewis - Shop Manager Challenge

atcq9876 opened the pull request.

atcq9876 committed:
Create files, design tables, create dbs, tables and seeds

atcq9876 committed:
Finish design recipe

atcq9876 committed:
Update design recipe, create class and spec files

atcq9876 committed:
Add test for item_repo #all method

atcq9876 committed:
Add test for and implement method #create(item)

atcq9876 committed:
Add tests for and implement methods #all and  #create(order): order_repository

atcq9876 committed:
Add tests for and implement 3 of 4 app conditions

atcq9876 committed:
Write test for and implement final condition (creating a new order)

atcq9876 committed:
Update order_repository's #update method to make it also update the join table

atcq9876 committed:
Update README.md

atcq9876 committed:
Update README.md

atcq9876 committed:
Update order_repository's #update method to make it decrement item quantity when an order is created

Cecily-R commented on lib/order.rb:
```diff
@@ -0,0 +1,7 @@
+class Order
+  attr_accessor :id, :customer_name, :item_id, :date, :items
```
I wouldn't necessarily know what the attribute ':items' was if you hadn't explained. Maybe it's worth adding a comment here just to clarify its function.

atcq9876 committed:
Update order_repository.rb

Update #create(order) method to decrement item quantity when an order is created

atcq9876 committed:
Merge branch 'main' of github.com:atcq9876/shop-manager-challenge

atcq9876 committed:
Resolve Git merge error: Update order_repository's #update method to make it decrement item quantity when an order is created

Cecily-R commented on items_orders_many_to_many_tables_recipe.md:
```diff
@@ -0,0 +1,155 @@
+Two Tables (Many-to-Many) Design Recipe Template
```
I went about this as one-to-many (and got confused). I like how you have gone with a more logical approach as the one-to-many tables structure doesn't quite fit this brief.

Cecily-R commented on spec/item_repository_spec.rb:
```diff
@@ -31,39 +29,30 @@ def reset_tables
-
-
 
+    it "raises an error if a non-Item object is passed as an argument" do
```
I like how you have put in an error message to catch any incorrect inputs.

atcq9876 committed:
Update README.md

atcq9876 committed:
Update README.md

atcq9876 committed:
Update README.md
